modules/shield/handle_msg.go

```diff
@@ -0,0 +1,188 @@
+package shield
+
+import (
+	"fmt"
+	"strconv"
+
+	shieldtypes "github.com/certikfoundation/shentu/v2/x/shield/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/forbole/bdjuno/v3/types"
+	juno "github.com/forbole/juno/v3/types"
+)
+
+// HandleMsg implements modules.MessageModule
+func (m *Module) HandleMsg(index int, msg sdk.Msg, tx *juno.Tx) error {
+	if len(tx.Logs) == 0 {
+		return nil
+	}
+
+	switch cosmosMsg := msg.(type) {
+	case *shieldtypes.MsgCreatePool:
+		return m.HandleMsgCreatePool(index, tx, cosmosMsg)
+
+	case *shieldtypes.MsgUpdatePool:
+		return m.HandleMsgUpdatePool(tx, cosmosMsg)
+
+	case *shieldtypes.MsgPausePool:
+		return m.HandleMsgPausePool(tx, cosmosMsg)
+
+	case *shieldtypes.MsgResumePool:
+		return m.HandleMsgResumePool(tx, cosmosMsg)
+
+	case *shieldtypes.MsgWithdrawRewards:
+		return m.HandleMsgWithdrawRewards(tx, cosmosMsg)
+
+	case *shieldtypes.MsgWithdrawForeignRewards:
+		return m.HandleMsgWithdrawForeignRewards(tx, cosmosMsg)
+
+	case *shieldtypes.MsgDepositCollateral:
+		return m.HandleMsgDepositCollateral(tx, cosmosMsg)
+
+	case *shieldtypes.MsgWithdrawCollateral:
+		return m.HandleMsgWithdrawCollateral(tx, cosmosMsg)
+
+	case *shieldtypes.MsgPurchaseShield:
+		return m.HandleMsgPurchaseShield(tx, cosmosMsg)
+
+	case *shieldtypes.MsgUpdateSponsor:
+		return m.HandleMsgUpdateSponsor(tx, cosmosMsg)
+
+	case *shieldtypes.MsgStakeForShield:
+		return m.HandleMsgStakeForShield(tx, cosmosMsg)
+
+	case *shieldtypes.MsgUnstakeFromShield:
+		return m.HandleMsgUnstakeFromShield(tx, cosmosMsg)
+
+	}
+
+	return nil
+}
+
+// HandleMsgCreatePool allows to properly handle a MsgCreatePool
+func (m *Module) HandleMsgCreatePool(index int, tx *juno.Tx, msg *shieldtypes.MsgCreatePool) error {
+
+	// Get create pool event
+	createPoolEvent, err := tx.FindEventByType(index, "create_pool")
+	if err != nil {
+		return fmt.Errorf("error while getting create pool event: %s", err)
+	}
+
+	// Get pool ID
+	poolIDStr, err := tx.FindAttributeByKey(createPoolEvent, shieldtypes.AttributeKeyPoolID)
+	if err != nil {
+		return fmt.Errorf("error while getting pool ID from MsgCreatePool: %s", err)
+	}
+
+	// Convert pool ID
+	poolID, err := strconv.ParseUint(poolIDStr, 10, 64)
+	if err != nil {
+		return fmt.Errorf("error while parsing pool ID to uint64: %s", err)
+	}
+
+	pool := types.NewShieldPool(
+		poolID, msg.Shield[0].Amount, msg.Deposit.Native, msg.Deposit.Foreign, msg.Sponsor,
```

Comment:
`msg.Shield[0].Amount`: do we want to store all coins in shield amount or just 1 type of coin is sufficient?

Reply:
I think there is only one coin available in shield coz `shield` and `shield_limit` are both `Int` [values in Pool](https://github.com/ShentuChain/shentu/blob/v2.3.2/x/shield/types/shield.pb.go#L112)